README.md

```diff
@@ -1,39 +1,132 @@
 
-* **GitHub** - all our code is hosted here. We use Pull Requests and do code reviews for those. Everyone on the team reviews PRs. It's expected that you write quality code with automated tests. Once a PR is reviewed and accepted, the person who opened the PR should merge it and delete the branch.
+### Things We value
 
-* **Continuous Integration** - we use Semaphore for our CI needs. Everytime we push to a branch, our test suite runs on Semaphore.
+- Clean code, we want you to show us the best code you can do
+- If you are familiar with TDD, BDD or any testing process, please show us your skills
 
-* **Slack** - we mostly communicate asynchronously between the development team and with other teams. This is our chat tool. Abuse it.
+## Solution
 
-* **Pivotal Tracker** - our products' backlog is managed in Pivotal Tracker. Once we estimate all known user-stories, the sprint backlog is automatically built. Each team member will select a user-story to work from that sprint backlog and will update the needed tasks' status. This way we keep focused until the end of the sprint.
+My solution is a Node.js service exposing an API on port `8080` with the following endpoints:
+- `GET /posts` 
+    - returns a list of posts in the expected order in JSON format
+- `GET /posts/:post_id` 
+    - returns the post with the specified ID
+    - returns error if post with specified ID is not found
+- `POST /posts` 
+    - create new post
+    - expects parameter `title` in body
+    - returns the newly created post
+- `PUT /upvote/:post_id` or `PUT /posts/:post_id/upvote`
+    - upvote specified post
+- `PUT /downvote/:post_id` or `PUT /posts/:post_id/downvote`
+    - downvote specified post
 
-* **Support tasks** - We are paranoid about providing top notch support to our customers. We have a dedicated customer support team working full-time communicating with clients. Our development team works very closely with the customer support team delivering happiness to our clients.
+### Technologies
+
+The following are the main technologies used:
+- Node.js
+- TypeScript
+- Express
+- Joi
+- Knex
+- Jest
+- PostgreSQL
+- Docker
+
+### Decisions and Assumptions
+
+Some decisions were made regarding architecture, database schema or functionality during the development of the solution. This is a list of the most notable ones.
+
+#### User authentication is out of scope
+
+I felt user authentication was out of scope of this challenge but since the challenge mentions my application has users I assumed all endpoints required the user to be authenticated to access. In order to simulate this all endpoints require the presence of a header named `rupeal-user` which represents the user accessing the endpoint. If this header is not present the service returns `401` error.
```

Comment:
Nice! Would you change this approach for a production environment? If yes, describe the solution

Reply:
I feel like this is a good approach even for production since it allows micro-services to care only about their functionality. In order for this to work there would have to be a "gateway" of sorts verifying each request and adding the required header(s) before forwarding the requests to the responsible service.

Reply:
So the gateway would be responsible for the users authentication? And if it's a valid user, it adds the header to the request, right?

Reply:
Yes, that is my suggestion. The other services would only need to check if the header was present and act accordingly.